### ISSUE TITLE
Bug in Version 1.7.3?

### ISSUE THREAD

**Issue by Engerim** [FriendsOfSymfony/FOSRestBundle#1255]
Since the scope attribute was removed from the service, the ParamFetcher does not init the params property again for the sub request, so if an exception is triggered inside the paramFetcher instead of rendering the exception response the paramfetcher throws the same exception again

**Comment by stof** [comment]
Anyway, removing the request scope was broken, as the params are specific to each request. The ParamFetcher needs to be refactored to be able to live in the container scope.
@Ener-Getick can you check this as you worked on the Symfony 3 compat ?

**Comment by Engerim** [comment]
a potential fix is to reset the params if a new controller is set in the paramfetcher

**Comment by stof** [comment]
this would not work. It would mean that after the subrequest, the main request would get the params of the subrequest instead of its own params

**Comment by Ener-Getick** [comment]
Yeah didn't think about this when I refactored the ``ParamFetcher`` to support ``RequestStack``...
This bug is probably also present in master as the ``initParams`` function hasn't been refactored.

**Comment by Ener-Getick** [comment]
What about saving the last request used in the param fetcher and change ``ParamFetcher::getParams()`` to:
```php
public function getParams() 
{
    $request = $this->getRequest();
    if ($this->lastRequest !== $request) {
        $this->initParams();
        $this->lastRequest = $request;
    }
    return $this->params;
}

**Comment by Ener-Getick** [comment]
Is #1263 an appropriate bug fix for you ?

**Comment by Ener-Getick** [comment]
Fixed in #1263.
Thank you!
